tmvis.js

```diff
@@ -985,12 +1007,33 @@ TimeMap.prototype.supplyChosenMementosBasedOnHammingDistanceAScreenshotURI = fun
   // Assuming foreach is faster than for-i, this can be executed out-of-order
   this.mementos.forEach(function (memento,m) {
     var uri = memento.uri
-      uri = uri.replace("id_/http","/http");
+
+
+      //  to replace /12345678912345id_/ to /12345678912345/
+      var regExpForDTStr = /\/\d{14}id_\// // to match something lile /12345678912345id_/
+      var matchedString = uri.match(regExpForDTStr)
+      if(matchedString != null){
+        uri = uri.replace(matchedString[0],(matchedString[0].toString().replace("id_",""))) // by default only the first occurance is replaced
+      }
+
+      //uri = uri.replace("id_/http","/http"); //replaced by above code segment
+
+
+
+
     // ConsoleLogIfRequired("Hamming distance = "+memento.hammingDistance)
     if (memento.hammingDistance < HAMMING_DISTANCE_THRESHOLD  && memento.hammingDistance >= 0) {
       // ConsoleLogIfRequired(memento.uri+" is below the hamming distance threshold of "+HAMMING_DISTANCE_THRESHOLD)
       memento.screenshotURI = null
-      var filename = 'timemapSum_' + memento.hammingBasisURI.replace("id_/http","/http").replace(/[^a-z0-9]/gi, '').toLowerCase() + '.png'  // Sanitize URI->filename
+
+      var regExpForDTStr = /\/\d{14}id_\// // to match something lile /12345678912345id_/
+      var matchedString = memento.hammingBasisURI.match(regExpForDTStr)
+      if(matchedString != null){
+        memento.hammingBasisURI = memento.hammingBasisURI.replace(matchedString[0],(matchedString[0].toString().replace("id_",""))) // by default only the first occurance is replaced
```

Comment:
Here as well...

You might want to abstract this repeated logic in two functions.